var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -99,6 +100,30 @@ class PyTorch(PythonPackage, CudaPackage):
     conflicts('+redis', when='@:1.0')
     conflicts('+zstd', when='@:1.0')
     conflicts('+tbb', when='@:1.1')
+    conflicts('cuda_arch=none', when='+cuda+caffe2',
+              msg='Must specify CUDA compute capabilities of your GPU, see '
+              'https://developer.nvidia.com/cuda-gpus')
+    conflicts('cuda_arch=52', when='@1.3.0:+cuda+caffe2',
+              msg='This version of Torch only supports compute capabilities '
+              '>= 5.3')
+    conflicts('cuda_arch=50', when='@1.3.0:+cuda+caffe2',
+              msg='This version of Torch only supports compute capabilities '
+              '>= 5.3')
+    conflicts('cuda_arch=35', when='@1.3.0:+cuda+caffe2',
+              msg='This version of Torch only supports compute capabilities '
+              '>= 5.3')
+    conflicts('cuda_arch=32', when='@1.3.0:+cuda+caffe2',
+              msg='This version of Torch only supports compute capabilities '
+              '>= 5.3')
+    conflicts('cuda_arch=30', when='@1.3.0:+cuda+caffe2',
+              msg='This version of Torch only supports compute capabilities '
+              '>= 5.3')
```

Comment:
Don't we need a conflict for `cuda_arch=20 @1.3.0`? Also, you can probably store these `msg` in a string and re-use them instead of repeating them.

Reply:
I now see a conflict exists with a cuda >= 9 depends statement with cuda_arch=20. I know that cuda_arch=20 works with 0.4 so that needs to be sorted out.

Reply:
```
depends_on('cuda@9:', when='@1.1:+cuda', type=('build', 'link', 'run'))
```
conflicts with the following from CudaPackage:
```
depends_on('cuda@:8', when='cuda_arch=20')
```

Reply:
I am kind of stuck here. I can not checkout the sources of 0.4.0 and 0.4.1 to test them. I get asked for my github username and password but then I get the following:"
```console
remote: Repository not found.
fatal: repository 'https://github.com/NervanaSystems/nervanagpu.git/' not found
Clone of 'https://github.com/NervanaSystems/nervanagpu.git' into submodule path 'third_party/nervanagpu' failed
==> Fetching from [git] https://github.com/pytorch/pytorch.git at tag v0.4.0 failed.
==> Error: FetchError: All fetchers failed for spack-stage-py-torch-0.4.0-djgjzjpsxivvsp3skf6x7pgkpmsowkqc

/opt/packages/gpjohnsn/spack/lib/spack/spack/package.py:1110, in do_fetch:
       1107                raise FetchError("Will not fetch %s" %
       1108                                 self.spec.format('{name}{@version}'), ck_msg)
       1109
  >>   1110        self.stage.create()
       1111        self.stage.fetch(mirror_only)
       1112        self._fetch_time = time.time() - start_time
       1113

```
Any conflicts directive that I set for cuda_arch=20 is superseded by the following:
```console
$ spack install py-torch%gcc@4.8.5 cuda_arch=20 ^cuda@7.5.18
==> Warning: gcc@4.8.5 cannot build optimized binaries for "broadwell". Using best target possible: "haswell"
==> Error: Conflicting conditional dependencies for spec

        py-torch@1.4.0%gcc@4.8.5~binary+caffe2+cuda cuda_arch=20 +cudnn~distributed~fbgemm~ffmpeg~gloo~leveldb~lmdb~magma~miopen+mkldnn+nccl~nnpack~opencv+openmp~qnnpack~redis~tbb~test~zstd arch=linux-centos7-haswell

Cannot merge constraint

        cuda@9:

into

        cuda@7.5:8
```
even if I set the cuda version.

Reply:
To be clearer, what I am trying to show above is that any spec with py-torch@1.1: will not be able to process conflicts directives involving cuda_arch=20. Another example:
```console
$ spack install py-torch cuda_arch=20                                                       
==> Error: Conflicting conditional dependencies for spec

        py-torch@1.4.0%gcc@8.3.0~binary+caffe2+cuda cuda_arch=20 +cudnn~distributed~fbgemm~ffmpeg~gloo~leveldb~lmdb~magma~miopen+mkldnn+nccl~nnpack~opencv+openmp~qnnpack~redis~tbb~test~zstd arch=linux-centos7-broadwell

Cannot merge constraint

        cuda@9:

into

        cuda@7.5:8
```
That is due to the dependency spec conflicts for cuda_arch=20 mentioned in https://github.com/spack/spack/pull/14619#discussion_r370762260.

Reply:
I committed a change that gets the conflicts set based on what I remember from a build of 0.4.0 from quite a while ago.

Reply:
The `nervanagpu` bug will be fixed by #14370.